src/views/Pools/components/CakeVaultCard/VaultCardActions/HasSharesActions.tsx

```diff
@@ -13,55 +15,43 @@ interface HasStakeActionProps {
   pool: Pool
   stakingTokenBalance: BigNumber
   stakingTokenPrice: number
-  userInfo: VaultUser
-  setLastUpdated: () => void
 }
 
-const HasSharesActions: React.FC<HasStakeActionProps> = ({
-  pool,
-  stakingTokenBalance,
-  stakingTokenPrice,
-  userInfo,
-  setLastUpdated,
-}) => {
+const InlineBalance = styled(Balance)`
+  display: inline;
+`
+
+const HasSharesActions: React.FC<HasStakeActionProps> = ({ pool, stakingTokenBalance, stakingTokenPrice }) => {
   const { pricePerFullShare } = useGetVaultPublicData()
+  const { userShares } = useGetVaultUserInfo()
   const { stakingToken } = pool
-  const { cakeAsBigNumber, cakeAsDisplayBalance } = convertSharesToCake(userInfo.shares, pricePerFullShare)
+  const { cakeAsBigNumber, cakeAsNumberBalance } = convertSharesToCake(userShares, pricePerFullShare)
 
-  const stakedDollarValue = formatNumber(
-    getBalanceNumber(cakeAsBigNumber.multipliedBy(stakingTokenPrice), stakingToken.decimals),
-  )
+  const stakedDollarValue = getBalanceNumber(cakeAsBigNumber.multipliedBy(stakingTokenPrice), stakingToken.decimals)
 
   const [onPresentTokenRequired] = useModal(<NotEnoughTokensModal tokenSymbol={stakingToken.symbol} />)
 
   const [onPresentStake] = useModal(
-    <VaultStakeModal
-      stakingMax={stakingTokenBalance}
-      pool={pool}
-      userInfo={userInfo}
-      stakingTokenPrice={stakingTokenPrice}
-      setLastUpdated={setLastUpdated}
-    />,
+    <VaultStakeModal stakingMax={stakingTokenBalance} pool={pool} stakingTokenPrice={stakingTokenPrice} />,
   )
 
   const [onPresentUnstake] = useModal(
-    <VaultStakeModal
-      stakingMax={cakeAsBigNumber}
-      pool={pool}
-      stakingTokenPrice={stakingTokenPrice}
-      userInfo={userInfo}
-      setLastUpdated={setLastUpdated}
-      isRemovingStake
-    />,
+    <VaultStakeModal stakingMax={cakeAsBigNumber} pool={pool} stakingTokenPrice={stakingTokenPrice} isRemovingStake />,
   )
 
   return (
     <Flex justifyContent="space-between" alignItems="center">
       <Flex flexDirection="column">
-        <Heading>{cakeAsDisplayBalance}</Heading>
-        <Text fontSize="12px" color="textSubtle">{`~${
-          stakingTokenPrice ? stakedDollarValue : <Skeleton mt="1px" height={16} width={64} />
-        } USD`}</Text>
+        <Balance fontSize="20px" bold value={cakeAsNumberBalance} decimals={5} />
```

Comment:
Same as in other place - should be returned to default 3 decimals I guess?

Reply:
We're treating ourselves to a coupla extra decimals here Cheems - the idea is that even users with smaller CAKE balances will get _almost_ continuous feedback with showing 5 decimals. We've tested with Bugs' chad CAKE balance and it fits 👌